### PR TITLE
Add Testing

### PR DESCRIPTION
**Pull request by Alabi12**
# Summary
This PR contains the following;
- [ ] Add RSpec as a gem to your project.

- [ ] Create a Request spec file for all your controllers.

- [ ] Make sure that for each action you check:

- If response status was correct.

- If a correct template was rendered.

- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+RSpec.describe PostsController, type: :request do
+  describe 'GET /posts' do
+    it 'renders index template with correct placeholder text for post' do
+      get '/posts'
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('Here are lists of posts')
+    end
+  end
```
Comment (brytebee) [inline spec/requests/posts_controller_spec.rb:11]: You are required to test the index page that renders all posts, which you have correctly done. 👏 
However, you are also required to also test the show page for a single post as well as a single user. Kindly add them.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Alabi12👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  describe 'GET /users/:id/posts/:id' do
+    it 'renders show template with correct placeholder text' do
```
Comment (Davidosky007) [inline spec/requests/posts_controller_spec.rb:14]: * Kindly add also a test that checks If the response body includes the correct placeholder text as its a requirement 
image

File: spec/requests/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    it 'renders index template with correct placeholder text for users' do
+      get '/users'
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('Here are lists of users')
+    end
+  end
+
+  describe 'GET /users/:id' do
+    it 'renders show template with correct placeholder text' do
```
Comment (Davidosky007) [inline spec/requests/users_controller_spec.rb:14]: * Kindly add also a test that checks If the response body includes the correct placeholder text as its a requirement 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Alabi12 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 

 ## Required Changes ♻️
* Nice Job working on the previous CR's request but you forgot to add one test, Kindly check add them as its a requirement 🙏 
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Robert, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**